users/serializers.py

```diff
@@ -110,6 +118,20 @@ def validate_email(self, value):
     def validate_username(self, value):
         """Validates the username field"""
         trimmed_value = value.strip()
+        openedx_username_taken = False
+        try:
+            openedx_username_taken = check_username_exists_in_edx(trimmed_value)
+        except (
+            HTTPError,
+            RequestsConnectionError,
+            EdxApiRegistrationValidationException,
+        ):
+            log.exception(
+                "edX username verification failure for username: %s",
+                trimmed_value,
+            )
+        if openedx_username_taken:
+            raise serializers.ValidationError(USERNAME_ALREADY_EXISTS_MSG)
```

Comment:
I realized this only when I was testing the PR, I think this block of code can/should be moved after we've validated the username against the regex that is [below](https://github.com/mitodl/mitxonline/pull/757/files#diff-4349bb2be4e3bfa15b93d2cf390f4df197e2a0b11e6edfe8abd3f09beec21273R135-R136).

So, Currently, We are bearing a cost of an extra edX username validation API call even if the predefined username fails the regex validation. When we register with a username e.g. `arslan@` (Invalid as per regex) we send a call to edX and regardless of edX API result we then validate the regex and if it fails, we send an error message to frontend. So, Changing the order will make the logic better and save an API call.